### ISSUE TITLE
Topics: what is gigasecond about?

### ISSUE THREAD

**Issue by kytrinyx** [exercism/bash#64]
In preparation for the launch of the Exercism redesign, we need to decide what topics to list for gigasecond.
See https://github.com/exercism/bash/issues/61 for context.

@ConstantlyLost, @canel-rom1, @dantiel, @deepbsd, @fbi1714, @fwten, @mkrehbs, @patbl, @rpalo - Having solved this, what would you say that gigasecond is about? Did you struggle with anything in particular? What did you learn?

**Comment by budmc29** [comment]
**About:** Date/time manipulation and using parameters.
**Tags:** date, time, string, transforming.
**Difficulty:** Easy.
**Core?:** I would vote yes.

The exercise feels easy and a good starting point, can be solved with 1 liner if the user just want to get it done fast, since there are no tests for incorrect inputs.

**Comment by deepbsd** [comment]
Hi Katrina,
I would say that gigasecond is more of an intermediate level task.  I think people who actually use bash every day tend to use it on sys admin type tasks, such as automating the creation of a dev environment or some sort of directory structure or searching for files or line entries matching a pattern.  Bash is great at that stuff but it can do so much more.  So I would place Gigasecond slightly beyond the comfort zone of someone who is just getting comfy with Bash.
I would say the exercise is about doing some basic math in Bash and also about using the built-in date and time functionality.  It's basically a walk into the past to see how it used to be done before perl and python et al.  :-D
Dave
 

    On Saturday, July 22, 2017 12:44 PM, Katrina Owen <notifications@github.com> wrote:
 

 In preparation for the launch of the Exercism redesign, we need to decide what topics to list for gigasecond.
See #61 for context.@ConstantlyLost, @canel-rom1, @dantiel, @deepbsd, @fbi1714, @fwten, @mkrehbs, @patbl, @rpalo - Having solved this, what would you say that gigasecond is about? Did you struggle with anything in particular? What did you learn?—

**Comment by kytrinyx** [comment]
> can be solved with 1 liner if the user just want to get it done fast, since there are no tests for incorrect inputs.

@budmc29 Are we missing tests? Or are you saying that we're not doing verification of the input format?

@deepbsd thanks so much for your input on this one. I would concur that this looks like a somewhat intermediate (difficulty 3 or 4, maybe).

**Comment by budmc29** [comment]
> @budmc29 Are we missing tests? Or are you saying that we're not doing verification of the input format?

Just the input format, all the tests assume that the user will pass a correct date.
That's not a bad thing, it helps to keep the exercise simple, because if you add invalid format, the difficulty would be increased a lot.

I still think the difficulty is easy, since you can get the exercise with it's current task done with a few google searches or man page readings.

**Comment by deepbsd** [comment]
>quote:I still think the difficulty is easy, since you can get the exercise with it's current task done with a few google searches or man page readings.  
>endquote

I guess this depends on whether exercism should appeal to the many new people installing Linux or using OSX for the first time.  These people may have never written a line of code before, but might just be discovering the power of the command line for the first time.  I would hope exercism could appeal to both the newest newbie as well as the most grizzled graybeard.  -Dave
 

    On Saturday, September 23, 2017 12:03 PM, Bud (Mugur) Chirica <notifications@github.com> wrote:
 

 
@budmc29 Are we missing tests? Or are you saying that we're not doing verification of the input format?
Just the input format, all the tests assume that the user will pass a correct date.
That's not a bad thing, it helps to keep the exercise simple, because if you add invalid format, the difficulty would be increased a lot.I still think the difficulty is easy, since you can get the exercise with it's current task done with a few google searches or man page readings.—

**Comment by budmc29** [comment]
@deepbsd what would you say the difficulty number is?
Also, what do you think about my proposed tags for this exercise?

**Comment by deepbsd** [comment]
@budmc29 I would say 4.  Probably a 3 on Linux.  Gnu date is not installed on OSX by default, so a user will probably need to install gdate using, say, Homebrew or another package manager.  Once that is done the solution is a one-liner.  

On the tags, maybe 'command substitution' instead of 'string transforming'?  What do you think?
 

    On Saturday, September 23, 2017 4:13 PM, Bud (Mugur) Chirica <notifications@github.com> wrote:
 

 @deepbsd what would you say the difficulty number is?
Also, what do you think about my proposed tags for this exercise?—

**Comment by budmc29** [comment]
> @budmc29 I would say 4.  Probably a 3 on Linux.  Gnu date is not installed on OSX by default, so a user will probably need to install gdate using, say, Homebrew or another package manager.  Once that is done the solution is a one-liner. 

Okay makes sense, didn't know about OSX not having Gnu date. We'll bump it for 4.

> On the tags, maybe 'command substitution' instead of 'string transforming'?  What do you think?

Yes, it sound good.

**Tags:** date, time, command substitution.

**Comment by kenden** [comment]
if we [do not want to depend on utilities](https://github.com/exercism/bash/issues/88), including `date`because they take different parameters on different platforms, then this exercise become very hard. 
I would even vote to remove it in this case.

**Comment by kotp** [comment]
`time` I think is available on OSX.  Is there something that is requiring gdate to be ran to get back a date?

**Comment by kenden** [comment]
@kotp The [example](https://github.com/exercism/bash/blob/master/exercises/gigasecond/example.sh#L4) fails with the macos version of `date`.

```
./gigasecond.sh '2011-04-25Z'
/bin/date: illegal option -- -
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```

Using `gdate`, it works fine.

**Comment by sjwarner-bp** [comment]
Closed via #128
